### ISSUE TITLE
The submodule.<name>.ignore config is not respected

### ISSUE THREAD

**Issue by sschuberth** [libgit2/libgit2#5304]
Note, this is coming from https://github.com/gitahead/gitahead/issues/289; as such I'm not using `libgit2` myself as a library.

### Reproduction steps

The [gitahead](https://github.com/gitahead/gitahead) client, which uses `libgit2`, lists submodules as "-dirty" even if they are configured to ignore untracked files via [submodule.&lt;name&gt;.ignore](https://git-scm.com/docs/git-config#Documentation/git-config.txt-submoduleltnamegtignore) being set to `ignore = untracked`.

### Expected behavior

I'd expect untracked files being ignored in the status for such submodules, just like "git status" / "git diff" on the command line do.

### Actual behavior

submodules with untracked files as listed as dirty despite the configuration to ignore untracked files.

### Version of libgit2 (release number or SHA1)

[libgit2 @ 16a2faf](https://github.com/stinb/libgit2/tree/16a2fafa341cbdfece21cec6a15ecd0f7bc43a5f)

### Operating system(s) tested

Ubuntu Linux 18.04 64-bit

**Comment by pks-t** [comment]
Yeah, true, we're completely missing support for this. I'll add the "up for grabs" label for now as I have some other things to tackle first before I got the time to fix the issue here.

Thanks for your report!
